src/IntegratesQueryBuilder.js

```diff
@@ -0,0 +1,50 @@
+import { Mixin } from 'mixwith'
+
+/**
+ * @needs MakesRequests
+ */
+export default Mixin((superclass) => class extends superclass
+{
+    beforeRequest(request) {
+        if (superclass.hasOwnProperty('beforeRequest')) {
+            superclass.beforeRequest(request)
+        }
+
+        if (!request.builder) {
+            return request
+        }
+
+        if (typeof request.builder !== 'function') {
+            throw new Error('Builder property must be a function.')
+        }
+
+        let queryBuilderModule
+
+        try {
+            queryBuilderModule = require('js-query-builder')
+        } catch (e) {
+            throw new Error('IntegratesQueryBuilder mixin requires "js-query-builder" dependency to be installed.')
+        }
+
+        const builder = queryBuilderModule.query()
+
+        request.builder(builder)
+
+        request.query = builder
```

Comment:
This is a nice integration of `js-query-builder` but I feel that we could probably do better. Something more laravel-like.

```js
let articles = Articles.query()
    .filter('age', 20)
    .sort('-created_at', 'name')
    .get()
```

We'd simply need to create a static `query` method that would return a new class `QueryBuilder`. This class needs to keep track of the model and have a `get` methods to send the accumulated request.

Note that both this approach and the current approach can be used together.

Anyway feel free to discard this if you'd rather leave things like this.

Reply:
I like your idea 👍